src/components/panel/panel.js

```diff
@@ -2550,3 +2702,28 @@ function getElement(el) {
       document.querySelector(el) : el;
   return angular.element(queryResult);
 }
+
+
+/**
+ * Moves an array element from one array position to another position.
+ * http://stackoverflow.com/questions/5306680
+ *     /move-an-array-element-from-one-array-position-to-another
+ * @param {number} oldIndex The original position of the array element.
+ * @param {number} newIndex The new position that the array element will move
+ *     to.
+ */
+Array.prototype.move = function(oldIndex, newIndex) {
+  while (oldIndex < 0) {
```

Comment:
Using a loop here and for `newIndex` seem like overkill. Couldn't you subtract the index and the array length and use that to figure out how many elements you need to add?